hermes/probe/create/create.go

```diff
@@ -35,85 +38,84 @@ import (
+// Read implements the standard Read interface, it returns a random stream of bytes generated by a pseudo-random generator.
 // Argument:
-//      buf: a byte slice serves as a buffer
+//      buf: a byte slice serves as an output buffer
 // Returns:
-//	n:  the size of the buffer buf following a Read operation
-//	err: error it should be nil when not done reading and io.EOF once the whole file contents have been read
+//		n:  the number of bytes read
+//		err: error it should be nil when not done reading and io.EOF once the whole file contents have been read
 func (r *randomFileReader) Read(buf []byte) (n int, err error) {
 	// check whether Read is done
-	if r.i >= r.size {
+	if r.i >= r.sizeBytes {
 		return 0, io.EOF
 	}
 	b := buf
-	if len(buf) > (r.size - r.i) {
-		// if the length of buffer is greater than the number of bytes left to read make the size of the buffer match the number of bytes left to read
-		b = buf[:(r.size - r.i)]
+	if len(buf) > (r.sizeBytes - r.i) {
+		// if the length of buffer is greater than the number of bytes left to read make the sizeBytes of the buffer match the number of bytes left to read
+		b = buf[:(r.sizeBytes - r.i)]
 	}
-	n, err = r.rand.Read(b) // n is now the length  of the buffer
+	// n is now the length  of the buffer
+	n, err = r.rand.Read(b)
 	if err != nil {
-		return n, err // in this case n = 0
+		// in this case n = 0
+		return n, err
 	}
 	r.i += n
 	return n, err
 }
 
-// NewReader creates a new randomFileReader and returns a pointer to it
-// Returns:
-//	*randomFileReader: a Reader that supports randomFiles. This Reader contains information about the file size in bytes as well as a pseudo-random bytes generator
-// Warning: The randomFileReader produced by NewReader is not thread safe.
-func (f *randomFile) NewReader() *randomFileReader {
+func (f *randomFile) newReader() *randomFileReader {
 	//  id will serve as a Seed and i - index of the currently read byte  will be set to 0 automatically in the returned reader
 	return &randomFileReader{
-		size: f.size,
-		rand: rand.New(rand.NewSource(int64(f.id))), // rand.NewSource() requires an int64 as an argument
+		sizeBytes: f.sizeBytes,
+		rand:      rand.New(rand.NewSource(int64(f.id))),
 	}
 }
 
 func newRandomFile(fileID int32, fileSize int) (*randomFile, error) {
-	if fileID < begin || fileID > end {
-		return nil, fmt.Errorf("the file id provided %d wasn't in the required range [%d, %d]", fileID, begin, end)
+	if fileID < minFileID || fileID > maxFileID {
+		return nil, fmt.Errorf("invalid argument: fileID = %d; want %d <= fileID <= %d", fileID, minFileID, maxFileID)
 	}
 	if fileSize > maxFileSizeBytes || fileSize <= 0 {
-		return nil, fmt.Errorf("invalid argument: randomFile.size = %d; want 0 < size <= %d", fileSize, maxFileSizeBytes)
+		return nil, fmt.Errorf("invalid argument: randomFile.sizeBytes = %d; want 0 < sizeBytes <= %d", fileSize, maxFileSizeBytes)
```

Comment:
Nit: Replace `randomFile.sizeBytes` with `fileSize`, as that's the name of the argument.

Reply:
arguments changed to sizeBytes as per previous comment